src/xwayland-unmanaged.c

```diff
@@ -119,7 +119,11 @@ unmanaged_handle_override_redirect(struct wl_listener *listener, void *data)
 static void
 unmanaged_handle_request_activate(struct wl_listener *listener, void *data)
 {
-	wlr_log(WLR_DEBUG, "request_activate not handled\n");
+	struct xwayland_unmanaged *unmanaged =
+		wl_container_of(listener, unmanaged, request_activate);
+	wlr_xwayland_surface_activate(unmanaged->xwayland_surface, true);
+	wlr_xwayland_surface_restack(unmanaged->xwayland_surface, NULL,
+				     XCB_STACK_MODE_ABOVE);
```

Comment:
Didn't think this through but you may be able to do something like this here:
```c
wl_list_remove(&unmanaged->link);
wl_list_insert(unmanaged->server->unmanaged_surfaces.prev, &unmanaged->link);
```
This should make it synchronized within Xwayland at least.
Regarding scene-graph I'll have a look at this later again.

Reply:
I like that; then I think we could just update the scene-graph so (not tested):
```
if (unmanaged->node) {
	wlr_scene_node_raise_to_top(unmanaged->node);
}
```

Reply:
yes, that sounds like a good solution

Reply:
Hmm, in testing, `unmanaged_handle_request_activate()` is called before `unmanaged_handle_map()` and so the `wl_list_remove()` call segfaults.

Reply:
v2 avoids the `unmanaged_handle_request_activate()` questions by not implementing it for now. Doing the restack in `unmanaged_handle_map()` seems more straightforward and is enough to solve the original issue :shrug: